src/widget/wtracktableview.cpp

```diff
@@ -828,20 +831,45 @@ void WTrackTableView::contextMenuEvent(QContextMenuEvent* event) {
     if (modelHasCapabilities(TrackModel::TRACKMODELCAPS_ADDTOCRATE)) {
         m_pCrateMenu->clear();
         CrateSelectResult allCrates(m_pTrackCollection->crates().selectCrates());
+
+        const QList<TrackId> trackIds = getSelectedTrackIds();
+
+        QHash<CrateId, QSet<TrackId>> currentCrates(m_pTrackCollection->crates().collectCrateTrackListOfTracks(trackIds));
         Crate crate;
         while (allCrates.populateNext(&crate)) {
-            auto pAction = std::make_unique<QAction>(crate.getName(), m_pCrateMenu);
+            auto pAction = std::make_unique<QWidgetAction>(m_pCrateMenu);
+            auto pCheckBox = std::make_unique<QCheckBox>(m_pCrateMenu);
+
+            pCheckBox->setText(crate.getName());
+            pCheckBox->setProperty("crateId", crate.getId().toInt());
+            pCheckBox->setEnabled(!crate.isLocked());
             pAction->setEnabled(!crate.isLocked());
-            m_crateMapper.setMapping(pAction.get(), crate.getId().toInt());
-            connect(pAction.get(), SIGNAL(triggered()), &m_crateMapper, SLOT(map()));
+            pAction->setDefaultWidget(pCheckBox.get());
+
+            // FIXME: we could use the tristate feature here and show grey in case of multi track selection
+            // in which not all track are in the crate
+            if(currentCrates.contains(crate.getId())) {
+                if(currentCrates[crate.getId()].size() != trackIds.size()) {
```

Comment:
Ok, this works, but it took me quite a while to find out how. 
I think this requires some comments.
currentCrates should also be renamed ... mmm 
or can we move the size compare if condition to collectCrateTrackListOfTracks ?

Reply:
My thought was, that the API maybe more useful in the future. Giving a list of tracks and getting a hash with the crates and the containing tracks. Maybe I was overthinking it. I changed the api to just count the tracks in the crate.